### PR TITLE
chore(TDKN-212): springboot 2

### PR DESCRIPTION
**Pull request by tduperron**
**What is the problem this Pull Request is trying to solve?**
 
The pull request change springboot and spring cloud versions.

**What is the chosen solution to this problem?**

* Set version on parent POM.
* Daikon-message as auto configuration to avoid bean overriding. Maybe must be splited in two auto-config.
* Sleuth must be fixed. :-1: 
* SCOPE_GLOBAL_SESSION removed in spring 5 :-1: 
* Daikon-service still required bean overriding :-1: 
**Link to the JIRA issue**

https://jira.talendforge.org/browse/TDKN-212 

**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[x] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/108/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/108/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/109/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/109/)

File: daikon-messages/messages-model-spring-support/src/main/java/org/talend/daikon/messages/spring/consumer/sleuth/SpringSleuthSettersConfiguration.java [hunk 1]
```diff
@@ -31,14 +37,14 @@ public CorrelationIdSetter correlationIdSetter(Tracer tracer) {
             public void setCurrentCorrelationId(String correlationId) {
                 long spanId = 0;
                 String name = "";
-                Span currentSpan = tracer.getCurrentSpan();
+                Span currentSpan = tracer.currentSpan();
                 if (currentSpan != null) {
-                    spanId = currentSpan.getSpanId();
-                    name = currentSpan.getName();
+                    spanId = currentSpan.context().spanId();
+                    // FIXME name = currentSpan.name();
```
Comment (sgandon) [inline daikon-messages/messages-model-spring-support/src/main/java/org/talend/daikon/messages/spring/consumer/sleuth/SpringSleuthSettersConfiguration.java]: is this PR a work in progress, meaning that you will fix this before the merge ?

File: daikon-multitenant/multitenant-spring-async/src/main/java/org/talend/daikon/multitenant/async/RequestContextPropagationConfiguration.java [hunk 2]
```diff
@@ -69,10 +69,11 @@ public SimpleRequestAttributes(RequestAttributes requestAttributes) {
                 this.requestAttributesMap.put(SCOPE_REQUEST, key,
                         requestAttributes.getAttribute(key, RequestAttributes.SCOPE_REQUEST));
             }
-            for (String key : requestAttributes.getAttributeNames(RequestAttributes.SCOPE_GLOBAL_SESSION)) {
-                this.requestAttributesMap.put(SCOPE_GLOBAL_SESSION, key,
-                        requestAttributes.getAttribute(key, RequestAttributes.SCOPE_GLOBAL_SESSION));
-            }
+            // FIXME SCOPE_GLOBAL_SESSION removed in spring 5
```
Comment (sgandon) [inline daikon-multitenant/multitenant-spring-async/src/main/java/org/talend/daikon/multitenant/async/RequestContextPropagationConfiguration.java]: same concern here as above, will the be fixed ?

File: daikon-spring/daikon-spring-mongo/src/main/java/org/talend/daikon/spring/mongo/MultiTenancyMongoDbFactory.java [hunk 3]
```diff
@@ -63,6 +68,22 @@ public PersistenceExceptionTranslator getExceptionTranslator() {
         return delegate.getExceptionTranslator();
     }
 
+    @Override
+    public DB getLegacyDb() {
+        final String databaseName = getDatabaseName();
+        return mongoClientProvider.get(tenantProvider).getDB(databaseName);
```
Comment (sgandon) [inline daikon-spring/daikon-spring-mongo/src/main/java/org/talend/daikon/spring/mongo/MultiTenancyMongoDbFactory.java]:
```suggestion
        LOGGER.debug("Using '{}' as Mongo database.", databaseName);
        return mongoClientProvider.get(tenantProvider).getDB(databaseName);
```

File: daikon-spring/daikon-spring-mongo/src/test/java/org/talend/daikon/spring/mongo/TestMultiTenantConfiguration.java [hunk 4]
```diff
@@ -30,14 +31,31 @@ public static void changeHost(String host) {
...
+    public MongoServer mongoServer() {
+        return initNewServer();
+    }
+
+    private MongoServer initNewServer() {
         // Applications are expected to have one MongoDbFactory available
-        return new SimpleMongoDbFactory(new Fongo("Standard Mongo").getMongo(), "standard");
+        MongoServer server = new MongoServer(new MemoryBackend());
+
+        System.out.println("Bind");
```
Comment (sgandon) [inline daikon-spring/daikon-spring-mongo/src/test/java/org/talend/daikon/spring/mongo/TestMultiTenantConfiguration.java]: Could you use a Logger please.

File: daikon-spring/daikon-spring-mongo/src/test/java/org/talend/daikon/spring/mongo/TestMultiTenantConfiguration.java [hunk 5]
```diff
@@ -80,27 +93,31 @@ public MongoClientProvider mongoClientProvider() {
...
+                mongoInstances.clear();
             }
 
             @Override
             public MongoClient get(TenantInformationProvider provider) {
                 final String name = provider.getDatabaseURI().getURI();
-                fongoInstances.computeIfAbsent(name, Fongo::new);
-                return fongoInstances.get(name).getMongo();
+                if (!mongoInstances.containsKey(name)) {
+                    System.out.println("Create server " + name);
```
Comment (sgandon) [inline daikon-spring/daikon-spring-mongo/src/test/java/org/talend/daikon/spring/mongo/TestMultiTenantConfiguration.java]: could you use loggers please.

File: daikon-spring/daikon-spring-mongo/pom.xml [hunk 6]
```diff
@@ -19,65 +16,52 @@
...
             <groupId>org.springframework.data</groupId>
             <artifactId>spring-data-mongodb</artifactId>
             <version>${spring-data-mongodb.version}</version>
+            <scope>compile</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>21.0</version>
```
Comment (sgandon) [inline daikon-spring/daikon-spring-mongo/pom.xml]: Could you add the version in the root pom please ?

File: daikon-messages/messages-model-spring-support/src/main/java/org/talend/daikon/messages/spring/consumer/sleuth/SpringSleuthSettersConfiguration.java (hunk 1 above)
Comment (fhuaulme) [reply daikon-messages/messages-model-spring-support/src/main/java/org/talend/daikon/messages/spring/consumer/sleuth/SpringSleuthSettersConfiguration.java]: yes

File: daikon-multitenant/multitenant-spring-async/src/main/java/org/talend/daikon/multitenant/async/RequestContextPropagationConfiguration.java (hunk 2 above)
Comment (fhuaulme) [reply daikon-multitenant/multitenant-spring-async/src/main/java/org/talend/daikon/multitenant/async/RequestContextPropagationConfiguration.java]: yes

**Comment by tduperron** [comment]
Moved to #434

**Pull request closed without merging**